packages/fxa-auth-server/lib/senders/emails/fluentLocalizer.ts

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { DOMLocalization } from '@fluent/dom';
+import { FluentBundle, FluentResource } from '@fluent/bundle';
+import { negotiateLanguages } from '@fluent/langneg';
+import { JSDOM } from 'jsdom';
+import * as fs from 'fs';
+import path from 'path';
+import { renderWithOptionalLayout, context } from './renderer';
+import availableLocales from 'fxa-shared/l10n/supportedLanguages.json';
+
+const OTHER_EN_LOCALES = ['en-NZ', 'en-SG', 'en-MY'];
+
+class FluentLocalizer {
+  constructor() {}
+
+  async localizeEmail(
+    templateName: string,
+    layoutName: string,
+    mailSubject: string,
+    variables: Record<any, any>,
+    acceptLanguage: string
+  ) {
+    const htmlDocument = renderWithOptionalLayout(
+      templateName,
+      { ...variables, ...context },
+      layoutName
+    );
+
+    const { document } = new JSDOM(htmlDocument).window;
+
+    const userLocales: Array<string> = [];
+
+    const parseLocales = acceptLanguage.split(',');
+    for (let locale of parseLocales) {
+      locale = locale.replace(/^\s*/, '').replace(/\s*$/, '');
+      const [lang, qual] = locale.split(';');
+      userLocales.push(lang);
+    }
+
+    const currentLocales = negotiateLanguages(
+      [...userLocales],
+      [...OTHER_EN_LOCALES, ...availableLocales],
+      {
+        defaultLocale: 'en-US',
+      }
+    );
+    const baseDir = path.join(__dirname, 'locales');
+
+    const mappedLocaleLinks = currentLocales.map((locale: string) => {
+      return `${baseDir}/${locale}.ftl`;
+    });
+
+    async function fetchResource(resource: string) {
+      try {
+        const response = fs.readFileSync(`${resource}`, 'utf8');
+        return response;
+      } catch (e) {
+        // We couldn't fetch any strings; just return nothing and fluent will fall
+        // back to the default locale if needed.
+        return '';
+      }
+    }
+
+    async function* generateBundles(resourceIds: Array<string>) {
+      let bundle = new FluentBundle(currentLocales);
+      for (const id of resourceIds) {
+        for (let resourceId of resourceIds) {
+          let source = await fetchResource(resourceId);
+          let resource = new FluentResource(source);
+          bundle.addResource(resource);
+        }
+      }
+
+      yield bundle;
+    }
+
+    const l10n = new DOMLocalization(mappedLocaleLinks, generateBundles);
+
+    l10n.connectRoot(document.documentElement);
+
+    await l10n.translateRoots();
+
+    const subject = await l10n.formatValue(mailSubject);
+    return {
+      html: document.documentElement.outerHTML,
+
+      text: document.documentElement
+        .querySelector('body')
+        ?.textContent?.replace(/(^[ \t]*\n)|(^[ \t]*)/gm, ''),
```

Comment:
What's this regex doing? Stripping HTML? At any rate text version is going to need some love -- not sure if we want to try and address this now or in a future PR:

![Screen Shot 2021-07-19 at 2 31 56 PM](https://user-images.githubusercontent.com/6392049/126202196-e10317db-866b-457e-af86-e0eeffb9177e.png)

Reply:
Yeah I was trying to replace all the white spaces and empty lines and it worked fine until those hidden headers were added. I'll see if I can gracefully remove those as well in this PR itself or we can create a follow-up PR on this!